src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+S3_ARTIFACT_DIR = "artifacts"
+DEVICE_PATH = "/var/lib/greengrass/device.pem.crt"
+PRIVATE_PATH = "/var/lib/greengrass/private.pem.key"
+CA_PATH = "/var/lib/greengrass/AmazonRootCA1.pem"
+JSON_FILE = "iot_setup_data.json"
+
+
+def install_greengrass_lite_from_source(commit_id: str, region: str):
+    # Get config
+    with open(JSON_FILE, 'r') as file:
+        data = json.load(file)
+
+    device_cert = data['DEVICE_CERT']
+    private_key = data['PRIVATE_KEY']
+    thing_name = data['THING_NAME']
+    thing_group_name = data['THING_GROUP_NAME']
+
+    # Download the source repo
+    download_result = _download_source(commit_id)
+    if not download_result:
+        return False
+
+    # Change the working dir
+    original_dir = os.getcwd()
+    os.chdir("aws-greengrass-lite")
+
+    # Set up an iot client
+    iot_client = client("iot", region_name=region)
+
+    try:
+
+        # Install build tools
+        install_result = _install_build_dependencies()
+        if not install_result:
+            return False
+
+        # Add user and group
+        add_result_ggcore = _add_user_and_group("ggcore", "ggcore")
+        add_result_gg_component = _add_user_and_group("gg_component",
+                                                      "gg_component")
+        if not add_result_ggcore or not add_result_gg_component:
+            return False
+
+        # Create several directories
+        dir1_result = _create_dir("/ggcredentials",
+                                  ownership=True,
+                                  flag='-R',
+                                  user="ggcore",
+                                  group="ggcore")
+        dir2_result = _create_dir("/var/lib/greengrass",
+                                  ownership=True,
+                                  user="ggcore",
+                                  group="ggcore")
+        dir3_result = _create_dir("/etc/greengrass")
+        if not dir1_result or not dir2_result or not dir3_result:
+            return False
+
+        # Build
+        build_result = _build_with_cmake()
+        if not build_result:
+            return False
+
+        # TES setup
+        tes_result = _tes_setup(device_cert, private_key)
+        if not tes_result:
+            return False
+
+        # Config setup
+        src_path = "docs/examples/sample_nucleus_config.yaml"
+        temp_path = "./config.yaml"
+        dest_path = "/etc/greengrass/config.yaml"
+
+        move_result1 = _copy_file(src_path, temp_path)
+        config_result = _modify_config(iot_client, thing_name, temp_path,
+                                       "ggcore", "ggcore")
+        move_result2 = _copy_file(temp_path, dest_path)
+        remove_result = _remove_file(temp_path)
+        if not config_result or not move_result1 or not move_result2 or not remove_result:
+            return False
+
+        # run nucleus
+        os.chmod('misc/run_nucleus', 0o775)
+        process = subprocess.run(['sudo', './misc/run_nucleus'],
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)